include/boost/thread/concurrent_queues/detail/sync_deque_base.hpp

```diff
@@ -11,6 +11,8 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
+#include <boost/bind.hpp>
```

Comment:
Do you mind to create another PR with this change only?

Reply:
I can do that. Which change specifically are you referring to?

Reply:
These ones https://github.com/boostorg/thread/pull/190/commits/5371b04154e73de9abbab1267b26676e9f748059

You can add also the shared_mutex changes https://github.com/boostorg/thread/pull/190/commits/c5ab2d6865e001c408de6b962a27d4f3f0304982, but we need to see what is the added value.